tomviz/EmdFormat.cxx

```diff
@@ -113,6 +113,28 @@ class EmdFormat::Private
                      H5T_STD_U32LE);
   }
 
+  bool attribute(const std::string& group, const std::string& name,
+                 std::string& value)
```

Comment:
I'm fairly sure this function needs some more error handling, but my Googling examples of reading string attributes from HDF5 files didn't show what types of errors you need to handle, just how to read the attributes.

Reply:
Does the API documentation not enumerate them?

Reply:
I've added a bit more error handling based on the API, but I was wondering about type issues.  Apparently the method of reading strings I used is only for variable length strings.... if the units string in the file is fixed-length it will fail.  But I can't figure out from the API how to test that, since the ways I have tried failed to read the files I know are right.

Reply:
I found a file where the strings are stored with a fixed width and the current approach segfaults.  Apparently when the strings are stored fixed width, you have to allocate memory to put them in and pass in a char*.  When the strings are stored non-fixed width, you pass in a char** and it allocates the memory for you.  But I haven't figured out the API to tell how the string is stored.